skpro/domains/domains.py

```diff
@@ -0,0 +1,165 @@
+        A tuple of values is eligible to be an interval if,
+            - it has length 2;
+            - the first element is strictly smaller than the second element;
+        """
+        if len(values) != 2:
+            raise ValueError(
+                f"Expected a tuple of length 2 for `values, bot got a tuple of length {len(values)}"
+            )
+        elif values[0] > values[1]:
+            raise ValueError(
+                f"The left-bound must be strictly smaller then the right-bound, but got {values[0]} and {values[1]}."
+            )
+        elif values[0] == values[1]:
+            raise Exception(
+                "The left-bound and the right-bound coincides. Use the class `Finite` to represent this set."
+            )
+        else:
+            return values[0], values[1]
+
+    def _resolve_parenthesis(self, parenthesis) -> str:
+        if self._left == (-float("inf"), -float("inf")):
+            return "()"
+        elif self._left == -float("inf"):
+            return "(" + self._PARENTHESIS[parenthesis][1]
+        elif self._right == float("inf"):
+            return self._PARENTHESIS[parenthesis][0] + ")"
+        else:
+            return self._PARENTHESIS[parenthesis]
+
+    def __contains__(self, item) -> bool:
+        if isinstance(item, (float, int)) is False:
+            return False
+        elif item == -float("inf"):
+            return item == self._left
+        elif item == float("inf"):
+            return item == self._right
+        elif self._parenthesis == "()":
+            return self._left < item < self._right
+        elif self._parenthesis == "[)":
+            return self._left <= item < self._right
+        elif self._parenthesis == "(]":
+            return self._left < item <= self._right
+        else:
+            return self._left <= item <= self._right
+
+    def __str__(self) -> str:
+        return f"{self._parenthesis[0]}{self._left}, {self._right}{self._parenthesis[1]}"
+
+    @property
+    def boundary(self) -> Tuple[float, float]:
+        return self._left, self._right
+
+
+class Finite(Discrete):
+    r"""
+    Class to represent finite subsets of the real line, i.e., sets of the form :math:`\{a_1, ..., a_n\}`,
+    where `a_1, ..., a_n \in \mathbb{R}`.
+    """
+
+    def __init__(self, values: Tuple[float, ...]):
+        self.values = self._validate_values(values=values)
+        super().__init__()
+
+    def _validate_values(self, values: Tuple[float, ...]) -> List[float]:
+        """The private method checks if a tuple of numbers is elegible to be a finite set."""
+        for value in values:
+            if not isinstance(value, (float, int)):
+                raise TypeError(f"Expected `float`or `int`, but got {type(value)}.")
+            if value in [-float("inf"), float("inf")]:
+                raise ValueError(f"Value {value} not accepted in finite set.")
+        return sorted(values)
```

Comment:
hm, how about `list(set(values))`? Tha talso makes them unique, and might help us later with intersection.

Reply:
You are right., because if we define a finite set like Finite((1, 1, 2, 2)), then we need to know just the points (1, 2). Moreover, I was not checking for repeated elements.

I update that.